scripts/http-jsonp-detection.nse

```diff
@@ -131,32 +134,33 @@ action = function(host, port)
 
     local target = tostring(r.url)
     target = url.parse(target)
-    target = target.path 
+    target = target.path
 
     -- First we try to get the response and look for jsonp endpoint there 
-    if r.response and r.response.body and r.response.status==200 then
+    if r.response and r.response.body and r.response.status and r.response.status==200 then
 
-      local report, status, func
+      local status, func, report
       status, func = checkjson(r.response.body)
 
       if status == true then
         --We have found JSONP endpoint
         --Put it inside a returnable table.
-        report = "JSONP endpoint found at " .. target .. ". Function name is " .. func
-        table.insert(output, report)
+        output_str = output_str .. "\n" .. target 
```

Comment:
Remember that format strings are preferred over concatenation. Specially if they are inside loops that construct strings.